java-frontend/src/main/java/org/sonar/java/se/checks/debug/DebugMethodYieldsOnInvocationsCheck.java

```diff
@@ -82,20 +81,25 @@ private static IdentifierTree getIdentifier(ExpressionTree methodSelect) {
 
   @Override
   public void checkEndOfExecution(CheckerContext context) {
-    reportAll();
+    reportAll(context);
   }
 
   @Override
   public void interruptedExecution(CheckerContext context) {
-    reportAll();
+    reportAll(context);
   }
 
-  private void reportAll() {
-    methodBehaviors.entrySet().forEach(entry -> {
-      MethodBehavior mb = entry.getValue();
-      IdentifierTree methodName = entry.getKey();
-      Set<Flow> flow = flowFromYield(mb, methodName);
-      reportIssue(methodName, String.format("Method '%s' has %d method yields.", methodName.name(), mb.yields().size()), flow);
+  private void reportAll(CheckerContext context) {
+    Map<Symbol.MethodSymbol, MethodBehavior> behaviors = new HashMap<>();
```

Comment:
I think this cache is not necessary, call to `CheckerDispatcher.peekMethodBehavior` is cheap, so no need to add another cache